emails/views.py

```diff
@@ -424,9 +418,34 @@ def _handle_reply(from_address, message_json):
             # TODO: send the user an email
             # that replies are a premium feature
             incr_if_enabled('free_user_reply_attempt', 1)
-            return HttpResponse(
-                "Rely replies require a premium account", status=403
+            return False
+    else:
+        # The From: is not a Relay user, so make sure this is a reply *TO* a
+        # premium Relay user
+        try:
+            [to_local_portion, to_domain_portion] = to_address.split('@')
+            address = _get_address(
+                to_address, to_local_portion, to_domain_portion
             )
+            user_profile = address.user.profile_set.first()
+            if not user_profile.has_premium:
+                incr_if_enabled('free_user_reply_attempt', 1)
+                return False
+        except (ObjectDoesNotExist):
+            return False
+    return True
+
+
+def _handle_reply(from_address, message_json, to_address):
+    mail = message_json['mail']
+    (lookup_key, encryption_key) = _get_keys_from_headers(mail['headers'])
+    reply_record = _get_reply_record_from_lookup_key(lookup_key)
+    address = reply_record.address
+
+    if not _reply_allowed(from_address, to_address, reply_record):
```

Comment:
We should verify the in-reply-to field instead of the `to_address` since the `to_address` will ONLY be`replies` on the local portion [here](https://github.com/mozilla/fx-private-relay/blob/main/emails/views.py#L300-L301).